src/gt4sd/s3.py

```diff
@@ -147,6 +159,46 @@ def sync_folder(
                 )
 
 
+def upload_file_to_s3(
+    host: str,
+    access_key: str,
+    secret_key: str,
+    bucket: str,
+    target_filepath: str,
+    source_filepath: str,
+    secure: bool = True,
+) -> None:
+    """
+    Sync the cache with the S3 remote storage.
+
+    Args:
+        host: s3 host address.
+        access_key: s3 access key.
+        secret_key: s3 secret key.
+        bucket: bucket name to search for objects.
+        target_filepath: path to save the objects in s3.
+        source_filepath: path to the file to sync.
+        secure: whether the connection is secure or not. Defaults
+            to True.
+
+    Raises:
+        S3SyncError: in case of S3 syncing errors.
+    """
+    try:
+        client = GT4SDS3Client(
+            host=host, access_key=access_key, secret_key=secret_key, secure=secure
+        )
+        logger.info("starting syncing")
+        client.upload_file(bucket, target_filepath, source_filepath)
+        logger.info("syncing complete")
+    except Exception:
+        logger.exception("generic syncing error")
+        raise S3SyncError(
+            "CacheSyncingError",
+            f"error in uploading path={target_filepath} with host={host} access_key={access_key} secret_key={secret_key} secure={secure} bucket={bucket}",
```

Comment:
In principle I would avoid logging secrets, for the open ones that is totally ok, but in case of custom deployments of storage it may exposed sensitive secrets.